### ISSUE TITLE
Incorrect 'no-unused-vars' error

### ISSUE THREAD

**Issue by jiangfengming** [ota-meshi/eslint-plugin-svelte#424]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.37.0

### What version of `eslint-plugin-svelte` are you using?

2.23.1

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports = {
  root: true,
  parser: '@typescript-eslint/parser',

  parserOptions: {
    extraFileExtensions: ['.svelte']
  },

  plugins: [
    '@typescript-eslint'
  ],

  extends: [
    'eslint:recommended',
    'plugin:svelte/recommended'
  ],

  env: {
    node: true,
    browser: true
  },

  overrides: [
    {
      files: ['*.svelte'],
      parser: 'svelte-eslint-parser',
      parserOptions: {
        parser: '@typescript-eslint/parser'
      }
    }
  ]
};

```
</details>

```svelte
<script lang="ts">
  export let close: (value: boolean) => void;

  close(true);
</script>

```

### What did you expect to happen?

No error occurred.

### What actually happened?

'value' is defined but never used. eslint [no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)

<img width="579" alt="image" src="https://user-images.githubusercontent.com/11750532/228597899-90864003-7e9c-4dbe-ab31-ac71af78dbbc.png">

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/jiangfengming/eslint-plugin-svelte-bug-report

**Comment by ota-meshi** [comment]
You should turn off the rule and use an alternative rule.
https://typescript-eslint.io/rules/no-unused-vars/
